default/python/universe_generation/starsystems.py

```diff
@@ -70,6 +70,16 @@ def name_planets(system):
         fo.set_name(planet, name)
 
 
+def can_have_no_planets(star_type):
+    """
+    Return True is system is allowed to have no planets.
+    This intentionally only checks the STAR_TYPE_MOD_TO_PLANET_SIZE_DIST so that
+    the following code forces most stars to have a planetary body so that in the
+    GUI the presence of a star implies a planet
+    """
+    return universe_tables.STAR_TYPE_MOD_TO_PLANET_SIZE_DIST[star_type][fo.planetSize.noWorld] != 0
```

Comment:
I added a comment to `universe_tables.py` to clarify that a positive value in the none column means a chance of empty systems of that type.

The check should be > 0 and I fixed that.

The check should not be > 20 or anything other than 0.  Change the value in `universe_tables.py` for neutron stars to 0 if there should be no chance of neutron stars having no planets.

An "engine" should deal with the "source material", in this case `universe_tables.py` honestly.    

I left it as is, because the chief thing that it complicates is the acquisition of neutronium, which is a late game problem, whose solution is part of the "fun".